### PR TITLE
feat: Adding isPalindrome for words and phrases. [Hacktoberfest]

### PR DESCRIPTION
**Pull request by madukm**
#### Description of Change

A program for checking if a given word or phrase is a palindrome.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/725"></a>

### CODE REVIEW

File: misc/word_palindrome.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+	free(temp);
+   	return 1;
+}
+
+/**
+ * @brief Test function
+ * @returns void
+ */
+void test(){
```
Comment (Panquesito7) [inline misc/word_palindrome.c:61]:
```suggestion
static void test(){
```

File: misc/word_palindrome.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	assert(isPalindrome("madam"));
+	assert(isPalindrome("Able was I ere I saw Elba"));
+	assert(!isPalindrome("Ilikeoranges"));
+	assert(!isPalindrome("Hacktoberfest"));
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
```
Comment (Panquesito7) [inline misc/word_palindrome.c:72]:
```suggestion
/**
 * @brief Main function
 * @param argc commandline argument count (ignored)
 * @param argv commandline array of arguments (ignored)
 * @returns
```

File: misc/word_palindrome.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+#include <ctype.h>
+
+/**
+ * @brief Check is a string is a palindrome.
+ * @param str string to check
+ * @returns 0 if is not a palindrome
```
Comment (Panquesito7) [inline misc/word_palindrome.c:25]:
```suggestion
 * @returns 0 if it's not a palindrome
```

File: misc/word_palindrome.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ * @returns 1 if is a palindrome
```
Comment (Panquesito7) [inline misc/word_palindrome.c:26]:
```suggestion
 * @returns 1 if it's a palindrome
```

File: misc/word_palindrome.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
+/**
+ *
```
Comment (Panquesito7) [inline misc/word_palindrome.c:2]:
```suggestion
```

File: misc/word_palindrome.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+ * @file
+ * @brief Check if a given word or phrase is a palindrome.
+ * @details
+ * A palindrome is a word, number, phrase, or other sequence of 
+ * characters which reads the same backward as forward, such as
+ * madam, racecar.
+ * This program will be able to check phrases, with words 
+ * separated by spaces, but can't deal with special characters.
+ * @author [Maria Eduarda Kawakami](https://github.com/madukm)
+ * @date 27 October, 2020
```
Comment (Panquesito7) [inline misc/word_palindrome.c:12]: I do not see any Doxygen sentence called `@date`.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/madukm/C/actions

File: misc/word_palindrome.c [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	assert(isPalindrome("Able was I ere I saw Elba"));
+	assert(!isPalindrome("Ilikeoranges"));
+	assert(!isPalindrome("Hacktoberfest"));
+}
+
+/**
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns
```
Comment (Panquesito7) [inline misc/word_palindrome.c:72]:
```suggestion
 * @returns 0 on exit
 */
```

File: misc/word_palindrome.c (hunk 6 above)
Comment (madukm) [reply misc/word_palindrome.c]: I fixed and pushed in my fork. Thank you for saying.

File: misc/word_palindrome.c [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+ * separated by spaces, but can't deal with special characters.
+ * @author [Maria Eduarda Kawakami](https://github.com/madukm)
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+#include <ctype.h>
```
Comment (Panquesito7) [inline misc/word_palindrome.c:18]: Add a brief description of what the headers/libraries add (see the example below).
```suggestion
#include <stdio.h>      ///
#include <stdlib.h>    ///
#include <assert.h>   /// for assert
#include <string.h>   ///
#include <ctype.h>   ///
```

Comment (madukm) [reply misc/word_palindrome.c]: Done. Thanks :)

File: misc/word_palindrome.c [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+ * This program will be able to check phrases, with words 
+ * separated by spaces, but can't deal with special characters.
+ * @author [Maria Eduarda Kawakami](https://github.com/madukm)
+ *
+ */
+
+#include <stdlib.h> //for malloc and free
+#include <assert.h> //for assert
+#include <string.h> //for strlen 
+#include <ctype.h> //for tolower
```
Comment (Panquesito7) [inline misc/word_palindrome.c:17]:
```suggestion
#include <stdlib.h> /// for malloc and free
#include <assert.h> /// for assert
#include <string.h> /// for strlen 
#include <ctype.h> /// for tolower
```

Comment (madukm) [reply misc/word_palindrome.c:17]: /// is standard? Should I always use it like that?

Comment (Panquesito7) [reply misc/word_palindrome.c]: Only for one-line variables and headers/libraries.
If you want to document some variables, do it like this:

```c++
uint32_t num1 = 0;  ///< brief documentation
```

**Pull request closed without merging**